### PR TITLE
Fix 'make clean USE_MKLDNN=1' will build mkldnn issue

### PR DESCRIPTION
**Pull request by jinhuang415**
## Description ##
Fix 'make clean USE_MKLDNN=1' will build MKLDNN issue, add target check in Makefile, only build mkldnn for default target(if not specify the target) or all target if USE_MKLDNN is defined.

### CODE REVIEW

**Comment by jinhuang415** [comment]
@zheng-da @pengzhao-intel

File: Makefile [hunk 1]
```diff
@@ -66,13 +66,22 @@ $(warning "USE_MKL2017 is deprecated. We will switch to USE_MKLDNN.")
 endif
 
 ifeq ($(USE_MKLDNN), 1)
+ifeq ($(MAKECMDGOALS),)
```
[the remark this answers is not in the document]
Comment (jinhuang415) [reply Makefile]: MAKECMDGOALS is a makefile builtin variable used to indicate the make target, if user input "make clean", MAKECMDGOALS will be 'clean', if user input "make all", MAKECMDGOALS  will be 'all', and default will be empty (we usually don't specify the target when make MXNet so MAKECMDGOALS will be empty and Makefile will find first target which is all to build).
So user don't need to set MAKECMDGOALS, it is set automatically by make.

[the remark this answers is not in the document]
Comment (zheng-da) [reply Makefile]: This is really a cool solution. Thanks. Learn some new stuff.

**Comment by zheng-da** [comment]
Just tried it again. It doesn't work if someone wants to build "make test".

**Comment by szha** [comment]
Why can't the mkl build logic be cleaned up and proper build target be added? Make is already built for managing such dependencies.

**Comment by jinhuang415** [comment]
@zheng-da still working on it due to limited bandwidth, will update if there is any progress.

**Pull request closed without merging**